src/app/pages/rpc-stx-call-contract/components/call-contract-details.layout.tsx

```diff
@@ -0,0 +1,124 @@
+import { useAsync } from 'react-async-hook';
+
+import { PayloadType, addressToString, fetchAbi } from '@stacks/transactions';
+import { SharedComponentsSelectors } from '@tests/selectors/shared-component.selectors';
+import { Stack, styled } from 'leather-styles/jsx';
+
+import { formatContractId } from '@leather.io/stacks';
+import { AddressDisplayer, Approver } from '@leather.io/ui';
+
+import { logger } from '@shared/logger';
+
+import { useStacksExplorerLink } from '@app/common/hooks/use-stacks-explorer-link';
+import { LoadingSpinner } from '@app/components/loading-spinner';
+import { useRpcRequestContext } from '@app/features/rpc-request/rpc-request.context';
+
+import type { RpcCallContractRequest } from '../rpc-stx-call-contract-container';
+import { FunctionArgumentList } from './function-argument-list';
+
+export function CallContractDetailsLayout() {
+  const {
+    rpcRequest: { unsignedTx },
+  } = useRpcRequestContext<RpcCallContractRequest>();
+  const { handleOpenStacksTxLink } = useStacksExplorerLink();
+
+  if (unsignedTx.payload.payloadType !== PayloadType.ContractCall)
+    throw new Error('Transaction is not a contract call');
+
+  const contractAddress = addressToString(unsignedTx.payload.contractAddress);
+  const contractName = unsignedTx.payload.contractName.content;
+  const functionName = unsignedTx.payload.functionName.content;
+
+  const contractAbi = useAsync(async () => {
+    try {
+      return await fetchAbi({ contractAddress, contractName });
+    } catch (error) {
+      logger.error(`Failed to fetch contract ABI for ${contractAddress}.${contractName}`, error);
+      return;
+    }
+  }, []);
+
+  if (contractAbi.loading)
+    return (
+      <Approver.Section>
+        <LoadingSpinner />
+      </Approver.Section>
+    );
+
+  if (!contractAbi.result) {
+    return (
+      <Approver.Section>
+        <Approver.Subheader>
+          <styled.span textStyle="label.01">Contract</styled.span>
+        </Approver.Subheader>
+        <styled.span textStyle="caption.01" color="red.action-primary-default">
+          Not a valid contract
+        </styled.span>
+      </Approver.Section>
+    );
+  }
+
+  const contractFunction = contractAbi.result.functions.find(fn => fn.name === functionName);
+
+  if (!contractFunction) {
+    logger.error(
+      `Attempting to call a function (\`${functionName}\`) that ` +
+        `does not exist on contract ${contractAddress}.${contractName}`
+    );
```

Comment:
this could run many times cause it's in render flow. Is this another illegal state?